### ISSUE TITLE
assertion failed: meta.is_event()

### ISSUE THREAD

**Issue by dinhani** [tokio-rs/console#553]
### What crate(s) in this repo are involved in the problem?

console-api

### What is the issue?

`console-api` is panicking when running in debug mode in this [`debug_assert!`](https://github.com/tokio-rs/console/blob/60bcf87537d414b21efbd84159207f9ecda5e914/console-api/src/common.rs#L48)

No idea if this is expected or not, but I identified that an event of kind `HINT` is reaching that point and triggering the assertion to fail.

It comes from the external crate `jsonrpsee-core`, so I cannot easily change it.

```rust
Metadata { 
  name: "enabled /home/renato/.cargo/registry/src/index.crates.io-6f17d22bba15001f/jsonrpsee-core-0.22.5/src/tracing.rs:13", 
  target: "jsonrpsee_core::tracing::client", 
  level: Level(Trace), 
  module_path: "jsonrpsee_core::tracing::client", 
  location: /home/renato/.cargo/registry/src/index.crates.io-6f17d22bba15001f/jsonrpsee-core-0.22.5/src/tracing.rs:13, 
  fields: {}, 
  callsite: Identifier(0x559831eadc48), 
  kind: Kind(HINT) 
}
```

### How can the bug be reproduced?

```rust
let client = HttpClientBuilder::default().build(jsonrpc_url).unwrap();
let _ = client.request::<bool, Vec<()>>("net_listening", vec![]).await
```

### Logs, error output, etc

```text
thread 'tokio' panicked at /home/renato/projects/console/console-api/src/common.rs:50:13:
assertion failed: meta.is_event()
```

### Versions

```text
├── console-subscriber v0.2.0
│   ├── console-api v0.6.0
```

### Possible solution

Maybe filtering out invalid metadata kinds before calling the function or using `TryFrom` instead.
The function is being called from [aggregator.rs](https://github.com/tokio-rs/console/blob/main/console-subscriber/src/aggregator/mod.rs#L449-L450) in `console-subscriber`.

### Would you like to work on fixing this bug?

yes

**Comment by hds** [comment]
@dinhani Thanks for your issue report!

I would say that this is definitely incorrect on `console-api` side, even in debug mode we should be more resilient.

This will only happen if the `console-subscriber` is receiving traces that aren't from Tokio, since in the Tokio instrumentation we only use events and spans, but if we're going to have this assertion in place, we should be filtering out non-event/non-span callsites and not sending them on to the aggregator (we won't do anything with them anyway).

If you're interested in working on this, I propose that we check that the metadata received by `register_callsite` is either an event or a span, and otherwise we return `Interest::never()` and don't do any more:

https://github.com/tokio-rs/console/blob/a0d20fd62df07470b6033524afc00a96d156aaa5/console-subscriber/src/lib.rs#L531

We could add a test for this behaviour, although it looks like our test set-up doesn't fail the test when the subscriber thread panics, which is weird. I'll look at that separately.

**Comment by dinhani** [comment]
I tested my project with the suggested change at `register_callsite` and it worked. There are no more panics.

I sent the suggested PR, but I still need to check how to test it properly in this project.

**Comment by hds** [comment]
This has been fixed in #554. Thanks for fixing your own report @dinhani!
